### PR TITLE
Add video controls ui

### LINKED ISSUE #67: Play/Pause button overlay
screen shot 2018-02-10 at 16 30 36

Inspiration:

- [instagram](https://www.instagram.com/p/BXHj-DllyYU/).
- [twitter](https://twitter.com/verge/status/957383241714970624).

### PR DESCRIPTION
**Pull request by breadadams**
Close #67

### CODE REVIEW

File: packages/react-microlink/src/components/Card/CardMedia/Video/index.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  render() {
+    const {cardSize, controls, video, image, muted, autoPlay, loading, loop} = this.props
+    const {playing, progress} = this.state
+
+    return (
+      <MediaWrap
+        className="microlink_card__media_video_wrapper"
+        cardSize={cardSize}
+        loading={loading}
+        onClick={e => {
```
Comment (Kikobeats) [inline packages/react-microlink/src/components/Card/CardMedia/Video/index.js]: Could be better extract the event handle into a named function, more readable.

Like `onClick={this.togglePlay}`

File: packages/react-microlink/src/components/Card/CardMedia/Video/index.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      >
+        <Video
+          className="microlink_card__media_video"
+          src={getUrlPath(video)}
+          poster={image}
+          muted={muted}
+          autoPlay={autoPlay}
+          loop={loop}
+          playsinline
+          innerRef={video => {
```
Comment (Kikobeats) [inline packages/react-microlink/src/components/Card/CardMedia/Video/index.js]: on one line: `innerRef={video => (this.videoEl = video)}`

File: packages/react-microlink/src/components/Card/CardMedia/Video/index.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            this.videoEl = video
+          }}
+          onTimeUpdate={() => this.videoPlaying()}
```
Comment (Kikobeats) [inline packages/react-microlink/src/components/Card/CardMedia/Video/index.js]: I think you don't need create a fat arrow for exec the function, just

`onTimeUpdate={this.videoPlaying}`

Comment (breadadams) [reply packages/react-microlink/src/components/Card/CardMedia/Video/index.js]: Not without needing to `.bind(this)` in the constructor

**Comment by breadadams** [comment]
Just realised we're showing the ⏯ button all the time (whilst playback is paused), whereas in the initial _inspiration_ screenshots, it's only shown when hovering the video. What do you think @Kikobeats?

**Comment by Kikobeats** [comment]
@breadadams I think that in this case show play button all time is good because you make easy for the user identify video content over image

**Comment by breadadams** [comment]
Perfect, I was leaning towards those reasons too 👍

**Comment by breadadams** [comment]
We need to get #84 merged and in here, to resolve the Travis warnings

**Comment by Kikobeats** [comment]
@breadadams because #84 is currently merged, do you think you can update the PR and we move this ahead?

File: packages/react-microlink/src/components/Card/CardMedia/Video/controls.js [hunk 4]
```diff
@@ -36,19 +41,17 @@ export const PlayButton = styled.div`
...
   bottom: 0;
   background: #e1e8ed;
-  height: 2px;
-  border-top-right-radius: 2px;
-  transition: opacity 0.3s ease;
+  height: ${({cardSize}) => getSize(PROGRESS_BAR_HEIGHT, cardSize)}px;
+  transition: opacity 0.15s ease-in-out;
 
-  .microlink_card:not(:hover) & {
+  div.microlink_card:not(:hover) & {
```
Comment (breadadams) [inline packages/react-microlink/src/components/Card/CardMedia/Video/controls.js]: We can't add `div`, cause the card could be rendered as _any_ html element @Kikobeats

File: packages/react-microlink/src/components/Card/CardMedia/Video/index.js [hunk 5]
```diff
@@ -1,10 +1,16 @@
...
 import styled from 'styled-components'
 
-import {getUrlPath} from '../../../../utils'
-
 import {PlayButton, ProgressBar} from './controls'
 import MediaWrap from '../wrap'
+import {getUrlPath} from '../../../../utils'
+
+const CustomMotion = styled(Motion)`
+div.microlink_card:not(:hover) & {
```
Comment (breadadams) [inline packages/react-microlink/src/components/Card/CardMedia/Video/index.js]: My previous comment about not using `div` would apply to this, @Kikobeats

Comment (Kikobeats) [reply packages/react-microlink/src/components/Card/CardMedia/Video/index.js]: actually this code is not necessary! removed

**Pull request merged**